### PR TITLE
Memory weight assignment from contract

### PR DESCRIPTION
**Pull request by janezpodhostnik**
Based on core contracts https://github.com/onflow/flow-core-contracts/pull/285

### CODE REVIEW

**Review by dsainati1 — APPROVED** [review]
thanks for doing this!

File: fvm/meter/weighted/meter.go [hunk 1]
```diff
@@ -199,9 +199,7 @@ func WithComputationWeights(weights ExecutionEffortWeights) WeightedMeterOptions
 func WithMemoryWeights(weights ExecutionMemoryWeights) WeightedMeterOptions {
 	if int(common.MemoryKindLast)-len(weights) != 1 {
 		return func(m *Meter) {
-			for kind, weight := range weights {
```
Comment (janezpodhostnik) [inline fvm/meter/weighted/meter.go:202]: @dsainati1 This was causing the tests to fail, as the the memory weights are assigned like this:
```
		memoryWeights:          DefaultMemoryWeights,
```

so this code caused changes to the `DefaultMemoryWeights`

File: engine/execution/computation/execution_verification_test.go [hunk 2]
```diff
@@ -270,8 +270,8 @@ func TestTransactionFeeDeduction(t *testing.T) {
 		checkResult   func(t *testing.T, cr *execution.ComputationResult)
 	}
 
-	txFees := uint64(1_0000)
-	fundingAmount := uint64(1_0000_0000)
+	txFees := uint64(1_000)
```
Comment (janezpodhostnik) [inline engine/execution/computation/execution_verification_test.go:273]: This was actually set to the wrong number.

File: fvm/context.go [hunk 3]
```diff
@@ -60,9 +60,9 @@ func newContext(ctx Context, opts ...Option) Context {
 const AccountKeyWeightThreshold = 1000
 
 const (
-	DefaultComputationLimit             = 100_000    // 100K
-	DefaultMemoryLimit                  = 10_000_000 // 10G
-	DefaultEventCollectionByteSizeLimit = 256_000    // 256KB
+	DefaultComputationLimit             = 100_000        // 100K
+	DefaultMemoryLimit                  = 10_000_000_000 // 10G
```
Comment (janezpodhostnik) [inline fvm/context.go:64]: comment was 10G which fits better in this case.

**Review by dsainati1 — APPROVED** [review]

**Pull request merged**